tests/rptest/tests/offset_for_leader_epoch_test.py

```diff
@@ -255,6 +258,7 @@ def all_offsets_valid():
 
         for epoch, partitions in epoch_offsets.items():
             offsets_for_leader_epoch = get_offsets_for_leader_epoch(epoch)
-            for p in offsets_for_leader_epoch:
-                assert p.partition in partitions
-                assert partitions[p.partition] == p.epoch_end_offset
+            for pid, offset in partitions.items():
+                assert pid in offsets_for_leader_epoch, f"Partition {pid} not found in offsets for leader epoch {epoch}"
+                assert offsets_for_leader_epoch[
+                    pid] == offset, f"Expected offset for partition {pid} in epoch {epoch} to be {offset}, but got {offsets_for_leader_epoch[pid]}"
```

Comment:
nit: this check is equivalent to `offsets_for_leader_epoch == offsets_for_leader_epoch | partitions`

Reply:
done.